### PR TITLE
Blog List Page Frontend Development

### PR DESCRIPTION
**Pull request by velezrenz**
Task Link: https://app.asana.com/0/1202718495875213/1202725278795542/f

**Definition of Done:**

- Follow UI Design

**Commands**

`cd frontend`
`npm start`

**Note**
Should be directed to home 
Click All Blogs ; should be directed to Blog list
Clicking the card should direct you to blog details

**Screenshot**

image

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -2,12 +2,14 @@ import React from 'react';
...
 	render() {
 		return (
 			<Routes>
 				<Route path='/' element={<HomePage />} />
-				<Route path='/blogs/:id' element={<BlogDetail />} />
+				<Route path='/posts/:slug' element={<BlogDetail />} />
+				<Route path='/posts' element={<BlogList />} />
 			</Routes>
 		);
 	}
```
Comment (kentlouise) [inline frontend/src/App.js]: Separate the imports for created and packages module.
Add a new line above the App component.
Change the class component to functional for consistency.
```javascript
import React from 'react';
import { Route, Routes } from 'react-router-dom';

import BlogDetail from './BlogDetail/BlogDetail';
import HomePage from './HomePage/HomePage';
import BlogList from './BlogList/BlogList';

const App = () => {
	return (
		<Routes>
			<Route path='/' element={<HomePage />} />
			<Route path='/posts/:slug' element={<BlogDetail />} />
			<Route path='/posts' element={<BlogList />} />
		</Routes>
	);
}

export default App;

```

File: frontend/src/BlogList/BlogContainer.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+			centered
+			fluid
+			href='/posts/:slug'
+			header={header}
+			meta={meta}
+			description={description}
+			extra={extra}></Card>
+	);
+};
+export default BlogContainer;
```
Comment (kentlouise) [inline frontend/src/BlogList/BlogContainer.js]: Arrange the code.
```javascript
import React from 'react';
import { Card, Icon } from 'semantic-ui-react';
import { useNavigate } from 'react-router-dom';

const BlogContainer = () => {
	const navigate = useNavigate();
	const header = ['Title of the article'];
	const meta = ['August 11, 2022'];

	const description = [
		'Leverage agile frameworks to provide a robust synopsis for high level overviews.',
	];

	const extra = (
		<a href='/' onClick={() => navigate('/')}>
			<Icon name='comment' />
			20 comments
		</a>
	);

	return (
		<Card
			centered
			fluid
			href='/posts/:slug'
			header={header}
			meta={meta}
			description={description}
			extra={extra}></Card>
	);
};

export default BlogContainer;

```

File: frontend/src/BlogList/BlogList.css [hunk 3]
```diff
@@ -0,0 +1,21 @@
+.ui.center.aligned.text.container {
+	font-style: normal;
+	font-size: larger;
+}
+h1 {
```
Comment (kentlouise) [inline frontend/src/BlogList/BlogList.css]: Add a new line.

File: frontend/src/BlogList/BlogList.css [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	color: red;
+	font-size: 50px;
+}
+.ui.center.container {
```
Comment (kentlouise) [inline frontend/src/BlogList/BlogList.css]: Add a new line.

File: frontend/src/BlogList/BlogList.css [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+	padding: 20px;
+}
+.ui.center.aligned.container {
```
Comment (kentlouise) [inline frontend/src/BlogList/BlogList.css]: Add a new line.

File: frontend/src/BlogList/BlogList.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PaginationList from './PaginationList';
+import NavBar from '../Navbar/NavBar';
+import BlogContainer from './BlogContainer';
+import './BlogList.css';
```
Comment (kentlouise) [inline frontend/src/BlogList/BlogList.js]: Separate the imports for created and packages module.
```javascript
import React from 'react';

import './BlogList.css';
import PaginationList from './PaginationList';
import NavBar from '../Navbar/NavBar';
import BlogContainer from './BlogContainer';
```

File: frontend/src/BlogList/BlogList.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+				<BlogContainer />
+				<BlogContainer />
+			</div>
+			<div className='ui center aligned container'>
+				<PaginationList />
+			</div>
+		</div>
+	);
+};
+export default BlogList;
```
Comment (kentlouise) [inline frontend/src/BlogList/BlogList.js]: Add a new line.

File: frontend/src/App.js (hunk 1 above)
Comment (velezrenz) [reply frontend/src/App.js]: Done: Changed to function component

File: frontend/src/BlogList/BlogList.js (hunk 6 above)
Comment (velezrenz) [reply frontend/src/BlogList/BlogList.js]: Done po

**Review by kentlouise — APPROVED** [review]
# Approved #8 
- Code looks good to me.

**Pull request merged**